plugins/base/src/main/java/edu/wpi/first/shuffleboard/plugin/base/layout/ListLayout.java

```diff
@@ -42,16 +44,58 @@ private void initialize() {
     retained = EasyBind.listBind(container.getChildren(), EasyBind.map(widgets, this::paneFor));
   }
 
-  private Pane paneFor(Component widget) {
-    BorderPane pane = new BorderPane(widget.getView());
+  private WeakHashMap<Component, Pane> panes = new WeakHashMap<>();
+
+  private Pane paneFor(Component component) {
+    if (panes.containsKey(component)) {
+      return panes.get(component);
+    }
+
+    BorderPane pane = new BorderPane(component.getView());
+    ActionList.registerSupplier(pane, () -> actionsForComponent(component));
     pane.getStyleClass().add("layout-stack");
-    EditableLabel label = new EditableLabel(widget.titleProperty());
+    EditableLabel label = new EditableLabel(component.titleProperty());
     label.getStyleClass().add("layout-label");
     BorderPane.setAlignment(label, Pos.TOP_LEFT);
     pane.setBottom(label);
+
+    panes.put(component, pane);
     return pane;
   }
 
+  private ActionList actionsForComponent(Component component) {
+    ActionList al = ActionList.withName(component.getTitle());
+    int i = widgets.indexOf(component);
+
+    if (i > 0) {
+      al.addAction("Move up", () -> {
+        widgets.remove(i);
+        widgets.add(i - 1, component);
+      });
+      al.addAction("Send to top", () -> {
+        widgets.remove(i);
+        widgets.add(0, component);
+      });
+    }
+
+    if (i < widgets.size() - 1) {
+      al.addAction("Move down", () -> {
+        widgets.remove(i);
+        widgets.add(i + 1, component);
+      });
+      al.addAction("Send to bottom", () -> {
```

Comment:
These should also belong to `SubsytemLayout`. You'll probably have to extract a common super class, since the only real differences are

1. SubsytemLayout sorts widgets by title, alphabetically
2. SubsystemLayout has a data source

And probably an abstract superclass of _that_ for taking care of creating panes etc. for widgets, since that code is very likely to be the same for all or most layouts

Reply:
Why? If SubsystemLayout sorts things by title, isn't it unwanted for users to be able to reorder widgets themselves?

Reply:
The layout doesn't keep the children sorted. It just kind of assumes they're sorted when deciding where to place a new child. Alphabetical sorting isn't likely to put components in a _logical_ ordering, which is why users should be able to customize the order